### PR TITLE
Check dimension equality within tolerance

### PR DESCRIPTION
**Pull request by deepanshs**
Example before

```py
dim_1 = cp.as_dimension([1, 10], type="monotonic", unit='s')
dim_2 = cp.as_dimension([1-1e-8, 10+1e-12], type="monotonic", unit='s')

dim_2 == dim_1
# False
```

With update
```py
dim_1 = cp.as_dimension([1, 10], type="monotonic", unit='s')
dim_2 = cp.as_dimension([1-1e-8, 10+1e-12], type="monotonic", unit='s')

dim_2 == dim_1
# True
```

### CODE REVIEW

File: csdmpy/dimension/linear.py [hunk 1]
```diff
@@ -70,7 +71,10 @@ def __eq__(self, other):
         other = other.subtype if hasattr(other, "subtype") else other
         if not isinstance(other, LinearDimension):
             return False
-        check = [getattr(self, _) == getattr(other, _) for _ in __class__.__slots__[:4]]
+
+        non_quantitative = ["reciprocal", "_complex_fft"]
+        quantitative = ["_count", "_increment"]
```
Comment (mccarthy677) [inline csdmpy/dimension/linear.py:76]: does checking the reciprocal dimension matter?

**Review by mccarthy677 — COMMENTED** [review]
looks good, the branch isn't showing up on my end so I can't try it out.  I've had issues with this especially after scaling, so like 2*[1,2,3] may not read as == to [2,4,6] because of differences on the order of 10^-12.  I think your test probably covers that situation, but it may be worth double-checking

**Comment by deepanshs** [comment]
> the branch isn't showing up on my end so I can't try it out.

- do a `git fetch upstream` to fetch new branches. 
- You can also see it in vscode, if you have `GitHub Pull Request` extension installed. 
- Or, you can also installed this branch on colab, `!pip install git+https://github.com/deepanshs/csdmpy.git@djs/dim_equality_with_tol`

**Comment by deepanshs** [comment]
@mccarthy677 is there any other questions/comment regarding this PR?

**Review by mccarthy677 — APPROVED** [review]

**Pull request merged**